coalib/results/TextPosition.py

```diff
@@ -15,12 +19,21 @@ def __init__(self, line: (int, None)=None, column: (int, None)=None):
         :param line:        The line in file or None, the first line is 1.
         :param column:      The column indicating the character. The first one
                             in a line is 1.
+        :param text:        The text containing the character.
+        :param position:    Position identifying the index of character in text.
         :raises TypeError:  Raised when line or columns are no integers.
         :raises ValueError: Raised when a column is set but line is None.
         """
         if line is None and column is not None:
             raise ValueError("A column can only be set if a line is set.")
 
+        if line and column and text and not position:
+            position = TextPosition.calc_position(text, line, column)
+
+        if (line and column) is None and (text and position) is not None:
+            line, column = TextPosition.calc_line_col(text, position)
```

Comment:
Based on gitter discussion, it would be better to do this lazily.

What I mean by "lazily" is that we should not do `calc_line_col` here. Rather, we should do it inside the `position` property. Basically `if self._position is None and (self.text is not None)` we calculate the position and return that.

This way, we don't do extra processing when `position` is not used at all.

@abhsag24 please confirm if you understand this